config/default/kustomization.yaml

```diff
@@ -25,13 +25,10 @@ resources:
 - ../internalcert
 # [PROMETHEUS] To enable prometheus monitor, uncomment all sections with 'PROMETHEUS'.
 #- ../prometheus
+# [METRICS] Expose the controller manager metrics service.
+- manager_metrics_service.yaml
```

Comment:
@ahg-g @anshuman-agarwala 

See that projects scaffolded with Kubebuilder now also have the default e2e tests generated: 
https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v4/test

The metrics can be used indeed to validate the reconciliations and etc

I would recommend you ensure that you also test it out in your tests.
See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/test/e2e/e2e_test.go#L166-L235

^ See here the tests to ensure the metrics endpoint
You might do not need the integration with prometheus. So, you could remove this block: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/test/e2e/e2e_test.go#L180-L183